### PR TITLE
[Technical] Fixed another assembly resolver error

### PR DESCRIPTION
**Pull request by duncanpMS**
Fixed corresponding unit tests (wasn't correctly testing the resolver)

### CODE REVIEW

File: Common/AssemblyResolver.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ private Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs a
                         (isFileName && string.Equals(Path.GetFileName(asm.Location), fileName,  StringComparison.OrdinalIgnoreCase))
                         ||
                         // ... otherwise compare against the full name
-                        (!isFileName || string.Equals(args.Name, asm.FullName, StringComparison.OrdinalIgnoreCase))
+                        (!isFileName && string.Equals(args.Name, asm.FullName, StringComparison.OrdinalIgnoreCase))
```
Comment (duncanpMS) [inline Common/AssemblyResolver.cs]: Incorrect logical comparison - need both conditions to be true.

File: Tests/CommonTests/AssemblyResolverTests.cs [hunk 2]
```diff
@@ -22,6 +22,19 @@ public class AssemblyResolverTests
...
+            // 1. Null logger
+            AssertException.Expect<ArgumentNullException>(() => new AssemblyResolver(null, new string[] { this.TestContext.TestDeploymentDir }));
+
+            // 2. Null paths
+            AssertException.Expect<ArgumentException>(() => new AssemblyResolver(new TestLogger(), null));
+
+            // 3. Empty paths
+            AssertException.Expect<ArgumentException>(() => new AssemblyResolver(new TestLogger(), new string[] { }));
+        }
+
```
Comment (duncanpMS) [inline Tests/CommonTests/AssemblyResolverTests.cs]: These are pretty low-value tests and I wouldn't normally write them for every class. However, I only found the most recent error by looking at the code coverage for the file and spotting a line expected to be covered wasn't. I added this test as it should take the coverage up to 100% as theses were the only other uncovered lines i.e. it'll make it simpler to see if the other cases have been covered without having to drill in to file.

File: Tests/CommonTests/AssemblyResolverTests.cs [hunk 3]
```diff
@@ -91,7 +104,7 @@ public void AssemblyResolver_ResolutionByFullAssemblyName_Succeeds()
             Assembly testAssembly = CompileSimpleAssembly("SimpleAssemblyByFullName.dll", testFolder, new TestLogger());
 
             // Act
-            Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("SimpleAssemblyByFullName, Version = 0.0.0.0, Culture = neutral, PublicKeyToken = null", testFolder);
+            Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("SimpleAssemblyByFullName, Version = 1.0.0.0, Culture = neutral, PublicKeyToken = null", testFolder);
```
Comment (duncanpMS) [inline Tests/CommonTests/AssemblyResolverTests.cs]: This test incorrectly passed before: we were passing getting the wrong version of the assembly back but didn't notice.

File: Tests/CommonTests/AssemblyResolverTests.cs [hunk 4]
```diff
@@ -154,18 +167,20 @@ public void AssemblyResolver_ResolutionByFileNameWithSpace_Succeeds()
...
             Assembly testAssembly = CompileSimpleAssembly("VersionAsm1.dll", testFolder, new TestLogger(), "2.1.0.4");
 
-            // Act
+            // 1. Search for a version that can be found -> succeeds
             Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("VersionAsm1, Version = 2.1.0.4, Culture = neutral, PublicKeyToken = null", testFolder);
-
-            // Assert
             AssertExpectedAssemblyLoaded(testAssembly, resolvedAssembly);
 
-            AssertAssemblyLoadFails("VersionAsm1, Version = 1.0.0.4, Culture = neutral, PublicKeyToken = null");
```
Comment (duncanpMS) [inline Tests/CommonTests/AssemblyResolverTests.cs]: This check wasn't exercising the product code so it didn't catch the product error.

**Comment by bgavrilMS** [comment]
LGTM

**Pull request merged**